runtime/src/rent_collector.rs

```diff
@@ -145,7 +159,9 @@ impl RentCollector {
         account: &impl ReadableAccount,
         filler_account_suffix: Option<&Pubkey>,
     ) -> RentResult {
-        if account.rent_epoch() > self.epoch {
+        if account.rent_epoch() == RENT_EXEMPT_RENT_EPOCH {
+            return RentResult::Exempt;
```

Comment:
@brooksprumo I'd rather this be `NoRentCollectionNow` so we don't set_rent_epoch again.